### PR TITLE
[monorepo] Update to use `x-forwarded-proto` and `x-forwarded-host` HTTP headers

### PR DESCRIPTION
**Pull request by TooTallNate**
This makes it work properly with `now dev` when it's running `next dev` instead of using a lambda.

### CODE REVIEW

File: monorepo/www/pages/index.js [hunk 1]
```diff
@@ -169,8 +169,9 @@ const Page = ({nows}) => <div className="container">
   </div>
 
 Page.getInitialProps = async ({req}) => {
-  const protocol = process.env.NOW_REGION === 'dev1' ? 'http' : 'https';
-  const baseUrl = `${protocol}://${req.headers.host}/api`
+  const protocol = req.headers['x-forwarded-proto']
```
Comment (styfle) [inline monorepo/www/pages/index.js]: Much better!

Comment (styfle) [reply monorepo/www/pages/index.js]: `x-forwarded-proto` should be available in both production and in dev correct?

**Comment by styfle** [comment]
@TooTallNate Should we document the headers here? https://zeit.co/docs/v2/routing/headers

**Comment by leo** [comment]
DMed @timothyis

**Pull request merged**